_posts/2020-06-10-cockpit-221.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Cockpit 221
+author: mvo
+date: 2020-06-10
+tags: cockpit linux
+slug: cockpit-221
+category: release
+summary: Cockpit with improved information protection and deprecated patternfly CSS API
+comments: 'true'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly. Here are the release notes from version 221.
+
+### Support for Cross-Origin-Resource-Policy
+
+Cockpit's web server now sets the
+[Cross-Origin-Resource-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cross-Origin_Resource_Policy_(CORP))
+header to `same-origin`. Web browsers use this header to prevent other
+web sites from loading individual HTML, JavaScript, or image resources
+from Cockpit via `<img>`, `<script>`, or similar tags. This is mostly
+a precaution -- currently there are no known vulnerabilities with
+including resources that way, but as this is not a legitimate use
+case, there is no reason to allow it.
+
+### Accounts: Hide some buttons when access is limited
+
+Instead of disabling the appropriate buttons when the logged in user
+doesn't have administrative access, they will now simply disappear.
+This is a general theme that we want to roll out all over Cockpit.
+The Overview, Storage, and Network pages have already been changed
+like this in earlier releases.
+
+### Developers: Importing "base1/patternfly.css" is deprecated
+
+This pre-compiled stylesheet will be dropped in the future in favor of
+projects shipping their own CSS. This API is not maintainable, as
+Cockpit cannot offer a PatternFly 3 API forever, and PatternFly 4 also
+changes quickly enough that one style sheet for all projects is not
+robust enough.
+
+The Cockpit plugins that are using only [PatternFly
+4](https://www.patternfly.org/v4/) should follow the example from
+[starter-kit](https://github.com/cockpit-project/starter-kit) on how
+to import [PatternFly 4](https://www.patternfly.org/v4/) stylesheets.
+g
+The Cockpit plugins which are still relying on [PatternFly
+3](https://www.patternfly.org/v3/) should follow the migration from
+the deprecated API to the new PatternFly stylesheet import approach as
+implemented in this [cockpit-podman
+commit](https://github.com/cockpit-project/cockpit-podman/commit/86fa1ee8b84ff50db0623a453de094b47a5bc2a4).
+
+### Try it out
+
+Cockpit 221 is available now:
+
+ * [For your Linux system](https://cockpit-project.org/running.html)
+ * [Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/221)
+ * [Fedora Rawhide](https://bodhi.fedoraproject.org/updates/FEDORA-2020-db8468a77a)
```

Comment:
Fedora infra plays along again. I just did the missing builds. Bodhi for F31: https://bodhi.fedoraproject.org/updates/FEDORA-2020-983ae499fa and F32: https://bodhi.fedoraproject.org/updates/FEDORA-2020-9c0127be8c

Reply:
I pushed the bodhi URLs to this PR.